### PR TITLE
Added new methods, retrieve and _get_sysmeta, and updated test

### PR DESCRIPTION
**Pull request by doulikecookiedough**
Added the retrieve method which calls _get_sysmeta method and returns the cid given the pid.
Also updated the "test_sysmeta_cid" test to "test_retrieve" which uses the retrieve method and asserts the same values.

### CODE REVIEW

File: src/hashstore/hashstore.py [hunk 1]
```diff
@@ -70,6 +70,19 @@ def _set_sysmeta(self, pid, sysmeta, obj_cid):
...
 
+    def _get_sysmeta(self, pid):
+        """Returns sysmeta content given persistent identifier (pid)"""
+        s_cid = self.hash_string(pid)
+        s_path = Path(self.abs_path(s_cid))
+        s_content = s_path.read_text()
+        cid = s_content[:64]
+        return cid
+
+    def retrieve(self, pid):
```
Comment (mbjones) [inline src/hashstore/hashstore.py:81]: The return value for retrieve should be the content of the cid, not the cid itself.

File: tests/test_hashstore.py [hunk 2]
```diff
@@ -71,15 +71,14 @@ def test_rel_path(pids, store):
     assert path.endswith("7052d7e166017f779cbc193357c3a5006ee8b8457230bcf7abcef65e")
 
 
-def test_sysmeta_cid(store):
+def test_retrieve(store):
```
Comment (mbjones) [inline tests/test_hashstore.py:74]: The test should test all of the files in the test harness, not just one.

**Review by mbjones — CHANGES_REQUESTED** [review]
Good start, @doulikecookiedough and thanks. I think retrieve should be the opposite of store and return the bytes of the content, not the cid. Also, it should work for both data vids and metadata cids. Finally, we may also want another method to support streaming the content rather than returning it in-memory.

File: tests/test_hashstore.py (hunk 2 above)
Comment (doulikecookiedough) [reply tests/test_hashstore.py]: Thanks Matt - I revised the test to check all the files in the test harness.

**Comment by doulikecookiedough** [comment]
Hi @mbjones, I updated the retrieve method to return the content of the cid as well the test to check all files in the test harness. 

**RE: Returning Data Vids:**
- My thought process here is that if we were to retrieve data vids, we would have to explicitly do so via a stream. I plan to review this interaction in issue_5 where we need to review the library/details.

**RE: Metadata CID Content:**
- The "_get_sysmeta" method returns the CID content - please let me know if this needs to be handled by the retrieve method instead.

**Comment by doulikecookiedough** [comment]
Hi @mbjones, I've updated my pull request based on our discussion and feedback from the Arc Dev session. 

**Notes:**
- get_sysmeta:
-- I feel that it makes more sense to return the sysmeta in a list with the header (hash + formatId) and content - which is how it is written/stored to begin with

- retrieve:
-- Returns sysmeta and io.BytesIO object/stream

**Comment by doulikecookiedough** [comment]
Hi @mbjones, I've re-factored the retrieve function to read file_obj in chunks (previously it was loading the whole file into memory before being converted into a ioBufferedIOBase object. I also added a new "retrieve_sysmeta" method which calls "_get_sysmeta".

After discussing with @artntek, we concluded that having a public method that specifically grab the sysmeta (without including the header content) would be the least disruptive. This way, users won't build dependencies on a method (_get_sysmeta) that could possibly change and/or require updates.

**Comment by doulikecookiedough** [comment]
Hi @mbjones, I've updated the PR. As per our discussion in Slack, "retrieve" should return a file handle to a stream (of a buffered IO class) - and it is the responsibility of the method caller to then process and close the connection.

**Comment by doulikecookiedough** [comment]
Hi @mbjones - Can you please review my PR and let me know if any further changes are needed?

**Comment by doulikecookiedough** [comment]
This pull request has been approved to merge into develop to prepare for alpha release and code review.

**Pull request merged**